app/models/manageiq/consumption/showback_event.rb

```diff
@@ -0,0 +1,38 @@
+class ManageIQ::Consumption::ShowbackEvent < ApplicationRecord
+  belongs_to :resource, :polymorphic => true
+
+  has_many :showback_charges, :dependent => :destroy
+  has_many :showback_buckets, :through   => :showback_charges
+
+  validates :start_time, :end_time, :resource, :presence => true
+  validate :start_time_before_end_time
+  validates :resource, :presence => true
+
+  serialize :data, JSON # Implement data column as a JSON
+
+  after_create :generate_data
+
+  self.table_name = "showback_events"
+
+  def start_time_before_end_time
+    errors.add(:start_time, "Start time should be before end time") unless end_time.to_i >= start_time.to_i
+  end
+
+  # return the parsing error message if not valid JSON; otherwise nil
+  def validate_format
+    JSON.parse(data) && nil if data.present?
+  rescue JSON::ParserError => err
+    err.message
+  end
+
+  def generate_data
+    self.data = {}
+    ManageIQ::Consumption::ShowbackUsageType.all.each do |measure_type|
+      next unless resource.type.ends_with?(measure_type.category)
```

Comment:
Do we need check on end of string here ?
is this useful for you to use this query instead of `ManageIQ::Consumption::ShowbackUsageType.all` skip `next unless` ?
`ManageIQ::Consumption::ShowbackUsageType.where.not(:category=>resource.type)`